src/Core/src/Stream/ResponseBodyResourceStream.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AsyncAws\Core\Stream;
+
+/**
+ * Stream a resource body.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class ResponseBodyResourceStream implements ResultStream
+{
+    /**
+     * @var resource
+     */
+    private $responseStream;
+
+    public function __construct($responseStream)
+    {
+        $this->responseStream = $responseStream;
+    }
+
+    public function __toString()
+    {
+        return $this->getContentAsString();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getChunks(): iterable
+    {
+        $pos = \ftell($this->responseStream);
+
+        try {
+            if (!\rewind($this->responseStream)) {
+                throw new \InvalidArgumentException('Failed to rewind the stream');
+            }
+
+            while (!\feof($this->responseStream)) {
+                yield \fread($this->responseStream, 64 * 1024);
+            }
```

Comment:
I wonder If I shouldn't ftell/fseek between each yield, to ensure that:
- reset the pointer to the original place, in case the user call something that need to read/write the same stream
- restore the pointer to the last read position (for the same reason)